### PR TITLE
Update templates for v3.5.0 release

### PR DESCRIPTION
**Pull request by humblec**
This PR update the required template changes for v3.5.0 release.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
closing this as it got opened against wrong branch

File: deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml [hunk 1]
```diff
@@ -111,8 +111,8 @@ spec:
             - name: socket-dir
               mountPath: /csi
         - name: csi-rbdplugin
-          # for stable functionality replace canary with latest release version
-          image: quay.io/cephcsi/cephcsi:canary
+          # for stable functionality replace v3.5.0 with latest release version
```
Comment (Madhu-1) [inline deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml]: why this change is required?

File: deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml [hunk 2]
```diff
@@ -164,8 +164,8 @@ spec:
             - name: ceph-config
               mountPath: /etc/ceph/
         - name: csi-rbdplugin-controller
-          # for stable functionality replace canary with latest release version
-          image: quay.io/cephcsi/cephcsi:canary
+          # for stable functionality replace v3.5.0 with latest release version
```
Comment (Madhu-1) [inline deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml]: same here

**Review by Madhu-1 — DISMISSED** [review]
where is upgrade documentation for 3.5 release?

**Comment by humblec** [comment]
> where is upgrade documentation for 3.5 release?

it will be a follow up PR @Madhu-1

**Comment by Madhu-1** [comment]
> > where is upgrade documentation for 3.5 release?
> 
> it will be a follow up PR @Madhu-1

@humblec usually during all releases the document changes are part of the template changes or before that not after the release.

**Comment by humblec** [comment]
> > > where is upgrade documentation for 3.5 release?
> > 
> > 
> > it will be a follow up PR @Madhu-1
> 
> @humblec usually during all releases the document changes are part of the template changes or before that not after the release.

@Madhu-1 the release is not rolled out, I was saying the document PR can be a follow up PR before/during the release.

**Comment by Madhu-1** [comment]
> > > where is upgrade documentation for 3.5 release?
> > 
> > 
> > it will be a follow up PR @Madhu-1
> 
> @humblec usually during all releases the document changes are part of the template changes or before that not after the release.

sent doc PR to devel first and backport it to release-3.5 and do a release? if we backport it after merging this PR. The docker image will be pushed 2 times.

**Comment by humblec** [comment]
> > > > where is upgrade documentation for 3.5 release?
> > > 
> > > 
> > > it will be a follow up PR @Madhu-1
> > 
> > 
> > @humblec usually during all releases the document changes are part of the template changes or before that not after the release.
> 
> sent doc PR to devel first and backport it to release-3.5 and do a release? if we backport it after merging this PR. The docker image will be pushed 2 times.

#2773 should cover it.

**Comment by humblec** [comment]
```    [91mJan 13 06:12:25.598: backend images not matching kubernetes resource count,image count 22 kubernetes resource count 21``` 

Jan 13 06:12:25.618: INFO: At 2022-01-13 06:10:01 +0000 GMT - event for rbd-32744: {kubelet minikube} FailedMount: MountVolume.MountDevice failed for volume "pvc-e7b62504-408e-43c1-ab69-b85da701ac8f" : rpc error: code = Internal desc = flatten in progress: flatten is in progress for image csi-vol-69831ce5-7437-11ec-b802-cee966cc1967

 https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.21/detail/mini-e2e-helm_k8s-1.21/2446/pipeline

**Comment by humblec** [comment]
@Madhu-1 the doc, mergify PRs are merged. ptal .. Thanks!

**Review by nixpanic — APPROVED** [review]
looks reasonable to me

**Review by yati1998 — APPROVED** [review]

**Pull request merged**